crates/aranya-daemon/src/daemon.rs

```diff
@@ -372,14 +377,18 @@ impl Daemon {
         // Sync in the background at some specified interval.
         let (send_effects, recv_effects) = tokio::sync::mpsc::channel(256);
 
+        // Create shared hello subscriptions for both server and syncer
+        let hello_subscriptions = Arc::new(Mutex::new(HashMap::new()));
+
         // Initialize the syncer
-        let (syncer, peers, conns, conn_rx) = Syncer::new(
+        let (mut syncer, peers, conns, conn_rx) = Syncer::new(
             client.clone(),
             send_effects,
             invalid_graphs,
             Arc::clone(&psk_store),
             server_addr,
             Arc::clone(&caches),
+            Arc::clone(&hello_subscriptions),
         )?;
```

Comment:
Or return `hello_subscriptions` from `Syncer::new()`. 
`let hello_subscriptions = Arc::new(Mutex::new(HashMap::new()));` doesn't need to be configured based on the daemon's config, so there isn't a reason to initialize it at this level.
```suggestion
        let (mut syncer, peers, conns, conn_rx, hello_subscriptions) = Syncer::new(
            client.clone(),
            send_effects,
            invalid_graphs,
            Arc::clone(&psk_store),
            server_addr,
            Arc::clone(&caches),
        )?;
```